### PR TITLE
ids: Verify at the end of the build the ids are ok

### PR DESCRIPTION
**Pull request by Spredzy**
With this commit, at the end of every build the entry
in /etc/passwd are compared to the entries in /root/ids.tables with
the verifiedids.py script.

### CODE REVIEW

File: build/verifyids.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+# Write gids list
+etc_passwd += "gids = {'" + user_list[0]['user'] + "': ('" + user_list[0]['gid'] + "', ''),"
+for user in user_list[1:-1]:
+  etc_passwd += " '" + user['user'] + "': ('" + user['gid'] + "', ''),"
+etc_passwd += " '" + user_list[-1]['user'] + "': ('" + user_list[-1]['gid'] + "', '')}"
+
+if ids_table == etc_passwd:
+  sys.exit(0)
+sys.exit(1)
```
Comment (EmilienM) [inline build/verifyids.py]: I would add debug here to have details of what is failing to easily add the UID/GID in ids.tables.

File: build/verifyids.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/python
+
+import sys
+
+# Parse /etc/passwd file and provide
+# put data into array
+passwd = open( "/etc/passwd", "r" )
```
Comment (EmilienM) [inline build/verifyids.py]: I would use "with_open" which automatically close the file after usage.

**Comment by EmilienM** [comment]
Some reviews inline, plus it would be great to have tests for this scripts, like we have for the other one.
Thanks

**Comment by Spredzy** [comment]
@EmilienM all remarks has been addressed. Thanks for reviewing.

File: build/common [hunk 3]
```diff
@@ -313,6 +313,11 @@ cleanup() {
         # Let's extract the target name from this variable
         CLEAN_CURRENT_TARGET=$(basename $CURRENT_TARGET | cut -d "." -f 1)
 
+        # Verify the IDS are correct
+        if [ -f "${dir}/usr/bin/verifyids.py" ] && ! do_chroot ${dir} /usr/sbin/verifyids.py; then
```
Comment (ErwanAliasr1) [inline build/common]: The /usr/bin/verifyids.py seems a typo isn't it ? 
Should be /usr/sbin/

File: build/test_verifyids.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
+#
+# Copyright (C) 2014 eNovance SAS <licensing@enovance.com>
```
Comment (ErwanAliasr1) [inline build/test_verifyids.py]: Please update Copyright date ;o)

**Comment by Spredzy** [comment]
@ErwanAliasr1 thanks for reviewing. Sorry for 2014, it might take me some time to update my inner clock ;)

**Comment by Spredzy** [comment]
On hold. Seeing with @morucci  about https://github.com/enovance/edeploy/pull/190

**Comment by EmilienM** [comment]
Dupplicated with https://github.com/enovance/edeploy/pull/190
We agreed to close this one. Thanks @Spredzy to have started this script :-)

**Pull request closed without merging**